browser/net/decentralized_dns_network_delegate_helper.cc

```diff
@@ -0,0 +1,106 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/net/decentralized_dns_network_delegate_helper.h"
+
+#include <vector>
+
+#include "net/base/net_errors.h"
+
+#include "brave/browser/brave_wallet/brave_wallet_service_factory.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_service.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
+#include "brave/components/brave_wallet/browser/eth_json_rpc_controller.h"
+#include "brave/components/decentralized_dns/constants.h"
+#include "brave/components/decentralized_dns/utils.h"
+#include "chrome/browser/browser_process.h"
+#include "content/public/browser/browser_context.h"
+
+namespace decentralized_dns {
+
+namespace {
+
+std::string GetValue(const std::vector<std::string>& arr, RecordKeys key) {
+  return arr[static_cast<size_t>(key)];
+}
+
+}  // namespace
+
+int OnBeforeURLRequest_DecentralizedDnsPreRedirectWork(
+    const brave::ResponseCallback& next_callback,
+    std::shared_ptr<brave::BraveRequestInfo> ctx) {
+  if (!ctx->browser_context || !IsDecentralizedDnsEnabled() ||
+      ctx->browser_context->IsOffTheRecord() || !g_browser_process) {
+    return net::OK;
+  }
+
+  if (IsUnstoppableDomainsTLD(ctx->request_url) &&
+      IsUnstoppableDomainsResolveMethodEthereum(
+          g_browser_process->local_state())) {
+    auto* service = BraveWalletServiceFactory::GetInstance()->GetForContext(
+        ctx->browser_context);
+    if (!service) {
+      return net::OK;
+    }
+
+    service->controller()->UnstoppableDomainsProxyReaderGetMany(
```

Comment:
not really clear: "many" of what? and the method doesn't explain anything

Reply:
also curious whether we consider the endpoint as trustworthy.. In theory, we'd better to follow Rule of 2: https://chromium.googlesource.com/chromium/src/+/master/docs/security/rule-of-2.md
probably something to consider during sec review

Reply:
It's calling getMany function defined by Unstoppable Domains's ProxyReader smart contract.

Reply:
We issue these calls to Unstoppable Domains smart contract via Infura to the Ethereum network.

Reply:
Added some more comments in 1cdb4f081decc67ac46ac53ed9bee83fcf3ebb34, and created security review issue.